### PR TITLE
CI: add travis config

### PR DESCRIPTION
**Pull request by smlng**
This PR does a couple of things to enable an initial CI check on applications:

~~1. it add the current/latest RIOT release as a submodule to this repository~~
~~2. RIOTBASE of all application Makefiles is adapted to point to the submodule~~
3. an initial travis.yml is added, that runs some static tests reusing scripts and config from the RIOT submodule
~~4. adds a patch for the `plot_rssi.py` of the spectrum-scanner to fix flake8 errors~~

~~The idea for the future would be that applications always work against the latest RIOT release and as part of the release process the submodule hash is updated every 3 month (or so). That way applications should work from release to release.~~

Note: I think someone with more GitHub rights than me has to enable Travis for this repo (@miri64?).

### CODE REVIEW

**Comment by kaspar030** [comment]
Please split. 1+2, 3, 4 are pretty distinct changes.

**Comment by smlng** [comment]
> Please split. 

ah damn, I knew someone might suggest that 😉 will do

**Comment by smlng** [comment]
requires #49 and #50

**Comment by miri64** [comment]
> Note: I think someone with more GitHub rights than me has to enable Travis for this repo (@miri64?).

I'll give it a try

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+dist: trusty
```
Comment (miri64) [inline .travis.yml]: Can you try to go to the new style of travis files we use in RIOT master?

https://github.com/RIOT-OS/RIOT/blob/eb6d9b052f6dc65ccf52b3d5a4eea35e54eb76b3/.travis.yml#L1-L14

**Comment by miri64** [comment]
No idea, how to trigger that build however. At the moment it is (rightly) complaining, that there is no `.travis.yml` in master :-/ https://travis-ci.org/RIOT-OS/applications/requests

File: .travis.yml (hunk 1 above)
Comment (smlng) [reply .travis.yml]: sure

**Comment by miri64** [comment]
@aabadie do you remember from when you re-integrated Travis back into the main repo?

**Comment by miri64** [comment]
> I think this will make it a lot easier for a release manager to update the submodule, w/o running tests on their own.

How? The applications are neither compiled nor tested on Travis with the current configuration.

**Comment by smlng** [comment]
> > I think this will make it a lot easier for a release manager to update the submodule, w/o running tests on their own.
> 
> How? The applications are neither compiled nor tested on Travis with the current configuration.

yeah, sorry was thinking further than this PR already 😄 If this is in, we can add compile test but for starters I think static tests are good

**Comment by smlng** [comment]
btw. I changed the style a bit and Travis also picked up on this

**Comment by miri64** [comment]
Yupp :-)

**Comment by smlng** [comment]
@miri64 you've the honours to hit the nice green button 😬

**Comment by miri64** [comment]
Done

**Comment by miri64** [comment]
Errrr somehow the submodule was reset to 2018.07 now. :-/ will fix

**Comment by smlng** [comment]
??? I rebases yesterday after the other PR was merged? Did Git(hup) mess up (again), had merge issues like that recently often...

**Comment by smlng** [comment]
mhm, looks like the other PR had the wrong commit 😞

**Comment by miri64** [comment]
Nope it was 2018.10 before. Now (included in your commit) is 2018.07 again.

**Comment by miri64** [comment]
See https://github.com/RIOT-OS/applications/pull/57

**Comment by smlng** [comment]
damn, you're right I oversaw the updated submodule and missed to remove that from the commit.

**Comment by miri64** [comment]
> on the other hand #57 provides a direct test of the enabled CI.

#57 passes, it only needs an ACK ;-)

**Pull request merged**